### PR TITLE
[Snowflake] format type options extensions for copy_into_location

### PR DESCRIPTION
**Pull request by YilangHe**
### Brief summary of the change made

Support more query commands for `copy_into_location`
Refer this [documentation](https://docs.snowflake.com/en/sql-reference/sql/copy-into-location.html#syntax)

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Comment by YilangHe** [comment]
Hi @tunetheweb,
Thank you so much for your previous review. Do you mind helping me to review this one?
I make some extensions for `formatTypeOption` of `copy_into_location` function in snowflake.

File: src/sqlfluff/dialects/dialect_snowflake.py [hunk 1]
```diff
@@ -4232,6 +4276,21 @@ class TableExpressionSegment(ansi.TableExpressionSegment):
     )
 
 
+class PartitionBySegment(BaseSegment):
+    """A `PARTITION BY` for `copy_into_location` functions."""
+
+    type = "partition_by_clause"
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_snowflake.py:4282]: This already exists in BigQuery but with a different type. Let's keep them consistent:

```suggestion
    type = "partition_by_segment"
```

Though we probably should be consistent with `PartitionClauseSegment` from ANSI too. But let's leave that for another day and at least not diverge to a third one.

**Review by tunetheweb — CHANGES_REQUESTED** [review]
Looks pretty good. One small change (which will require regenerating the YAMLs).

File: src/sqlfluff/dialects/dialect_snowflake.py (hunk 1 above)
Comment (YilangHe) [reply src/sqlfluff/dialects/dialect_snowflake.py]: Change pushed

**Review by tunetheweb — APPROVED** [review]

**Pull request merged**